seclang/parser.go

```diff
@@ -86,13 +86,16 @@ func (p *Parser) FromString(data string) error {
 			continue
 		}
 
-		if !inQuotes && line[lineLen-1] == '`' {
-			inQuotes = true
-		} else if inQuotes && line[0] == '`' {
-			inQuotes = false
+		// Looks for a line like "SecDataset test `". The backtick starts an action list.
+		// A line starting with # is a comment, therefore it must NOT be able to start an action list.
+		// The list will be closed only with a single "`" line.
+		if !inBackticks && line[0] != '#' && line[lineLen-1] == '`' {
```

Comment:
I think this can be easily fixed by moving [this line](https://github.com/corazawaf/coraza/pull/433/files#diff-f73e6b849d0fa747f9e706cbdb5ec76083f712530e21d64ea917cc37eb6347c5R104) upwards. When I originally read this code I thought comments could not include backticks and backticks can't include comments (maybe I am wrong) as described ni https://github.com/corazawaf/coraza/pull/412/commits/717e8c0f72c05b8d567755d27582ef7874530a2e#r966891278. So in this case:
1. if a comment includes backtick we skip it
2. if a backtick includes a comment it is not anymore a comment (please clarify @jptosso @piyushroshan)

Reply:
I changed now the fix to skipping any line starting with a `#`. Speaking about SecDataSet, comments inside backticks are possible and were just skipped right after. I can't find any other reason to start a line with a `#` inside backticks 🤔